test-framework/junit5/src/main/java/io/quarkus/test/junit/internal/XStreamDeepClone.java

```diff
@@ -0,0 +1,23 @@
+package io.quarkus.test.junit.internal;
+
+import com.thoughtworks.xstream.XStream;
+
+/**
+ * Super simple cloning strategy that just serializes to XML and deserializes it using xstream
+ */
+public class XStreamDeepClone implements DeepClone {
+
+    private final XStream xStream;
+
+    public XStreamDeepClone(ClassLoader classLoader) {
```

Comment:
I don't like this, jboss marshalling has some Cloner support that is likely more appropriate to use here. 

Really though if we had support for creating an object without calling a constructor you could just use reflection to recursively clone, but adding this support requires multi release jar support and would prevent Quarkus being build with JDK8. 

Ideally we would move non-constructor based instantiation into a mini library, and then do a simple reflection based clone. I probably won't get to look at this today, but I will likely have time later this week.

Reply:
Sure, that's obviously the right way to do things, but is it really warranted to spend the extra cycles for this use case?

Reply:
Not to mention the various corner cases that we would need look out for as opposed to this way we seems to "just work" - although I obviously agree that it's not the cleanest solution in the world :)

Reply:
Took a quick look at JBoss Marshalling and it's cloner support is just for serializable or cloneable objects

Reply:
In any case, if you think it best, I'll wait until the no-constructor library is in place and then update the PR to use it - just ping me about it :)

Reply:
From memory it has an interface that you can implement to decide if an object can be cloned, and you can use to just pretend everything is serialisable. Lots of classes are also not going to work with the XML based approach.

Thinking about it wouldn't XStream also require a no-arg constructor, so a reflection based approach would not be any worse?

Reply:
XStream doesn't call the constructor at all, see:

```
XStream is not calling the default constructor during deserialization.

This is, in fact, the same case as above. XStream uses the same mechanism as the JDK serialization. When using the enhanced mode with the optimized reflection API, it does not invoke the default constructor. The solution is to implement the readResolve or readObject as demonstrated with the last question.
```

from https://x-stream.github.io/faq.html.

So it should just work (and indeed in the test cases I added, the objects being passed around include ones with a no-args constructor)

Reply:
FWIW, I chose this strategy because it's also used in Powemock :)

Reply:
ok, lets just go with it then

Reply:
Cool :)